native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -2224,13 +2333,20 @@ static void remove_workers_nodes(proxy_server_conf *conf, apr_pool_t *pool, serv
 static void * APR_THREAD_FUNC proxy_cluster_watchdog_func(apr_thread_t *thd, void *data)
 {
     apr_status_t rv;
-    apr_pool_t *pool;
+    server_rec *s = main_server;
+    void *sconf = s->module_config;
+    proxy_server_conf *conf = (proxy_server_conf *)
+        ap_get_module_config(sconf, &proxy_module);
+    apr_pool_t *cached_pool, *pool;
+    apr_pool_create(&cached_pool, conf->pool);
```

Comment:
Is this pool suitable for our table allocations?  Or would there be a better means of managing the memory?

Reply:
In fact you are leaking the cached_pool each time the proxy_cluster_watchdog_func() is called. That WON'T work.

Reply:
Moved cached pool creation to proxy_cluster_child_init